### ISSUE TITLE
Release T5 `eta_fn` and `shape_fn` used in paper experiments

### ISSUE THREAD

**Issue by rpeloff-id** [google-research/jestimator#9]
Hi, thank you for this interesting work! Could you please release or provide an example for the `eta_fn` and `shape_fn` used for the Amos optimiser to train T5-large in the paper? I am assuming that these experiments were done using the model/network defined in this config from T5X: https://github.com/google-research/t5x/blob/main/t5x/examples/t5/t5_1_1/large.gin

**Comment by tianran** [comment]
Hi! Yes, the T5X experiments are conducted using that model definition, and because Amos optimizer has just been integrated into the T5X codebase, in order to use Amos you just need to add the following section into your gin config file:

```
from t5x import optimizers

# ------------------- Optimizer ------------------------------------------------
OPTIMIZER = @optimizers.amos()
optimizers.amos:
  learning_rate = @utils.create_learning_rate_scheduler()
  eta_fn = @eta_fn/amos_helper.params_fn_from_assign_map()
  shape_fn = @shape_fn/amos_helper.params_fn_from_assign_map()
  beta = 0.95
  momentum = 0.9
  clip_value = 684.  # sqrt(batch_size * target_length)
eta_fn/amos_helper.params_fn_from_assign_map:
  # In the value for r'.*/query/kernel', 64 is head_dim.
  assign_map = {r'.*/scale': 1.0, r'.*/embedding': 1.0, r'.*/relpos_bias/rel_embedding': 0.5, r'.*/query/kernel': 'sqrt(1/(SHAPE[0]*64))', r'.*/mlp/wo/kernel': 'sqrt(2/SHAPE[0])', r'.*/kernel': 'sqrt(1/SHAPE[0])'}
  eval_str_value = True
shape_fn/amos_helper.params_fn_from_assign_map:
  assign_map = {r'.*/kernel': '(1, SHAPE[1])', r'.*/embedding': '(SHAPE[0], 1)', r'.*': ()}
  eval_str_value = True

utils.create_learning_rate_scheduler:
  factors = 'constant'
  base_learning_rate = 0.01
```

Soon I will add some instructions from this repo as well.
Best!

**Comment by tianran** [comment]
Btw, if you are training on a different corpus, try replacing the base_learning_rate by 1/sqrt(N) where N is the number of batches in the training data. It should usually work out-of-the-box :)

**Comment by tianran** [comment]
↑ Oh and import the following if required :)

```
from jestimator import amos_helper
from t5x import utils
```

**Comment by rpeloff-id** [comment]
Hi, thank you for the great example! 😄 I am testing this on a different corpus so I will set the base learning rate to the corresponding number of batches. I find the disentangling of the learning rate into model and data dependent terms very appealing!
